portal-api/src/v1/controllers/utilisation-report-service/utilisation-report-download.controller.js

```diff
@@ -37,7 +43,7 @@ const getReportDownload = async (req, res) => {
     });
 
     if (bufferedFile.error) {
-      const errorMessage = `Failed to get utilisation report for download with _id '${_id}' from Azure Storage`;
+      const errorMessage = `Failed to get utilisation report for download with id '${id}' from Azure Storage`;
       console.error(errorMessage, bufferedFile.error);
       return res.status(500).send(errorMessage);
```

Comment:
Could we throw a custom error here instead of doing a console error and returning?
```
if (bufferedFile.error) {
      throw BlaError(id)
}
```
and then at the bottom we can keep our current handling, but add in a `console.error` log :) 

I quite like having all our return "at the bottom" and all our surprising returns (ie not 2xx or 3xx) in a catch block and handling them like errors -- what do you think (also very open to discuss other patterns!)?

Reply:
happy to leave as is as this ticket has scope creeped